src/Cli/dotnet/Program.cs

```diff
@@ -167,7 +177,7 @@ internal static int ProcessArgs(string[] args, TimeSpan startupTime, ITelemetry
                     bool generateAspNetCertificate =
                         environmentProvider.GetEnvironmentVariableAsBool("DOTNET_GENERATE_ASPNET_CERTIFICATE", defaultValue: true);
                     bool telemetryOptout =
-                      environmentProvider.GetEnvironmentVariableAsBool("DOTNET_CLI_TELEMETRY_OPTOUT", defaultValue: false);
+                      environmentProvider.GetEnvironmentVariableAsBool(TelemetryOptout, defaultValue: false);
```

Comment:
maybe avoid the indirection:
```suggestion
 #if NO_TELEMETRY_BY_DEFAULT
                      true; 
 #else
                      environmentProvider.GetEnvironmentVariableAsBool(TelemetryOptout, defaultValue: false);
#endif
```
and remove SetEnvironmentVariable call?

Reply:
That's getting closer but the polarity is still wrong. We want the non-conditional code to disable telemetry. This change is not doing that.

Reply:
```
 #if NO_TELEMETRY_BY_DEFAULT
                      true; 
 #else
```

https://github.com/dotnet/sdk/issues/24474 requests to make telemetry opt-in. That means: off by default, and enable through the envvar.

This suggestion unconditionally disables it. That doesn't match the request.

> We want the non-conditional code to disable telemetry.

Do you mean you'd like to 'invert' the preprocessor macro so it becomes `TELEMETRY_BY_DEFAULT` and the branches get swapped?

Reply:
Maybe I'm just having trouble with the logic. This is what I was expecting.

```suggestion
 #if MICROSOFT_ENABLE_TELEMETRY
                      environmentProvider.GetEnvironmentVariableAsBool(TelemetryOptout, defaultValue: false);
#else
                      true;
#endif
```

Reply:
>                       true;

This disables telemetry without making it possible to opt-in.

I think the request in https://github.com/dotnet/sdk/issues/24474 is to do this:

```cs
                      environmentProvider.GetEnvironmentVariableAsBool(TelemetryOptout, defaultValue:
#if MICROSOFT_ENABLE_TELEMETRY
                          false
#else
                          true;
#endif
                      );
```

Reply:
I see. I'd say that's the most minimal change we can make. Perhaps that's the right change to make first. In the fullness of time, I'd hope we could remove more of the telemetry system for the non-MS scenarios. However, if folks are happy with that change, that works for me.

Reply:
And another change so it also applies to:

https://github.com/dotnet/sdk/blob/ea8da05a744d8b1cc463c00543453635c0684cc1/src/WebSdk/Publish/Tasks/WebConfigTelemetry.cs#L34

Reply:
Nice find. Do you want to make that change?

Reply:
I would have liked one of the maintainers to chime in about their preference on how to handle the envvar.

I'll change it based on your input.

Reply:
@marcpopMSFT @baronfel @jkotas

Reply:
My ideal solution would centralize the access behind a function/member so that there was only one place we ever retrieved it.  Barring that the explicit read checks are my next-preferred solution. The half-baked idea being that the symmetry would inspire someone reading the code to centralize the access.

Reply:
Would that be a static somewhere? I think that's the wrong direction since it assumes a runtime check. ideally, there is no (telemetry-related) code that runs when `MICROSOFT_ENABLE_TELEMETRY` is not defined.